src/wfqc/metrics.py

```diff
@@ -1,81 +1,313 @@
 import igraph
+import numpy as np
 
-def sum_metric_citation(graph, workflow): # cocitation metric 
 
-    """
-    Calculates the cocitation  metric for a given workflow and a given citation (OBS not cocitation) graph
-    
-    Parameters
-    ----------
-    graph : igraph.Graph
-        Graph generated by igraph
-    workflow, list of tuples with pairs of strings TODO: how do I write this? does it matter I wont have this format later anyways? 
-        List of tuples of strings corresponding to the edges in the workflow. 
 
-    """
-    # List to  collect pairwise scores
-    score_list = [] # TODO: can predefine the list length, does not matter this is temporary?
+def get_node_ids(graph: igraph.Graph, key= "name"):
+    # connect names to igraph ids
+    if key == 'name':
+        return {v['name']:v.index for v in graph.vs}
+    elif key == 'index':
+        return {v.index:v['name'] for v in graph.vs}
+    else:
+        print("Not a valid key")
+        return {}
 
-    for pair in workflow:
-        cocite_score = 0
-        neighbors_of_first = set(graph.neighbors(pair[0]))
-        neighbors_of_second = set(graph.neighbors(pair[1]))
 
-        # Count number of common neighbours
-        common_neighbors = neighbors_of_first.intersection(neighbors_of_second)
-        cocite_score = len(common_neighbors)
-        score_list.append(cocite_score)
+def get_graph_edge_weight(graph, edge):
 
-    # Then sum the scores or perform any other desired calculation
-    # now normalising by WF length
-    # maybe call this one "support", since that is basically what we have. 
+    id_dict = get_node_ids(graph)
 
-    return sum(score_list)/len(score_list), score_list
+    if edge[0] not in graph.vs['name'] or edge[1] not in graph.vs['name']:
+        return None 
+    try:
+        source = edge[0]
+        target = edge[1]
+        weight = graph.es.find(_between=((id_dict[source],), (id_dict[target],)))['weight']
+    except:
+        weight = 0
```

Comment:
Is it a valid logic that in the case of a missing edge the method returns `None`, but in the case of an `Exception` the method returns `0`